source/test.ts

```diff
@@ -15,30 +15,30 @@ kava.suite('valid-directory', function (suite) {
 	suite('api', function (suite, test) {
 		test('valid', function (done) {
 			validate(paths.valid)
-				.then((path) => {
-					equal(path, paths.valid, 'path is valid')
+				.then((path: IsValidDirectory) => {
+					equal(path[0], true, 'path is valid')
+					done()
 				})
 				.catch((error) => {
-					equal(1, 2, 'path is invalid')
+					done(error)
 				})
-			done()
 		})
 		test('invalid', function (done) {
 			validate(paths.invalid)
-				.then((path) => {
-					equal(1, 2, 'path is valid')
+				.then((path: IsValidDirectory) => {
```

Comment:
`(path: IsValidDirectory)` -> `([isValid, invalidPaths])`

check that isValid is false, as you have done

but also check that invalidPaths.length is as expected

Reply:
There seems to be some issue with what readdir reports as invalid paths upon failure - it seems to report valid paths in this array as well - looking into it.